aspnetcore/signalr/hubs/sample/Hubs/ChatHub.cs

```diff
@@ -7,32 +7,52 @@ namespace SignalRChat.Hubs
 {
     public class ChatHub : Hub
     {
-        public async Task SendMessage(string user, string message)
+        #region HubMethods
+        public Task SendMessage(string user, string message)
         {
-            await Clients.All.SendAsync("ReceiveMessage", user,message);
+            return Clients.All.SendAsync("ReceiveMessage", user, message);
         }
 
         public Task SendMessageToCaller(string message)
         {
             return Clients.Caller.SendAsync("ReceiveMessage", message);
         }
 
-        public Task SendMessageToGroups(string message)
+        public Task SendMessageToGroup(string message)
         {
-            List<string> groups = new List<string>() { "SignalR Users" };
-            return Clients.Groups(groups).SendAsync("ReceiveMessage", message);
+            return Clients.Group("SignalR Users").SendAsync("ReceiveMessage", message);
         }
+        #endregion
 
+        #region HubMethodName
+        [HubMethodName("SendMessageToUser")]
+        public Task DirectMessage(string user, string message)
+        {
+            return Clients.User(user).SendAsync("ReceiveMessage", message);
+        }
+        #endregion
+
+        #region ThrowHubException
+        public async Task ThrowException()
```

Comment:
nit: remove async, this wouldn't compile as is.

Reply:
It'll compile, but with a warning (CS1998: This async method lacks 'await' operators and will run synchronously...). I originally intended to model an "async" exception here and using `async` ensures that a state-machine is used. We handle synchronous exceptions too though so I think I will switch it up to drop `async`.